### DISCUSSION TITLE
Modulation in .wav generation

### DISCUSSION THREAD

**Discussion by steenl** [pschatzmann/arduino-audio-tools#1048] [Q&A] [answered]
Hi Phil, can I get suggestion on next steps in taking I2S INMP443 streamed to a 5 second .wav file in ESP32-CAM?  Without the mic, the resulting .wav is header with zeros, and with mike the .wav file has nonzero content, but is 50% 0x00 or 0xFF, so I am getting some stream output but very noisy, so I am wondering if sampling is going "rail-to-rail".  Below are my configuration and output setup.  Could there be a configuration I missed, or should I try a different mic?
Appreciate your thoughts on this, thanks.

    auto cfg = i2sStream.defaultConfig(RX_MODE);
    cfg.i2s_format = I2S_STD_FORMAT; // or try with I2S_LSB_FORMAT
    cfg.bits_per_sample = 16;
    cfg.channels = 2;
    cfg.sample_rate = 8000; //44100;
    cfg.is_master = true;
     // this module needs a master clock if the ESP32 is master
    cfg.use_apll = false;  // try with yes
    cfg.buffer_size = 512;
    cfg.buffer_count = 2;
    cfg.pin_mck = 0; 
    cfg.pin_ws = 13;
    cfg.pin_data = 15;
    i2sStream.begin(cfg);
    // make sure that we have the correct channels set up
    EncodedAudioStream bitStream(&file, new WAVEncoder());
    bitStream.begin(cfg);
    StreamCopy copier(bitStream, i2sStream); // copy i2sStream to bitStream

[I] AudioTypes.h : 104 - sample_rate: 8000
[I] AudioTypes.h : 105 - channels: 2
[I] AudioTypes.h : 106 - bits_per_sample: 16
[I] I2SConfigESP32.h : 90 - rx/tx mode: RX_MODE
[I] I2SConfigESP32.h : 91 - port_no: 0
[I] I2SConfigESP32.h : 92 - is_master: Master
[I] I2SConfigESP32.h : 93 - sample rate: 8000
[I] I2SConfigESP32.h : 94 - bits per sample: 16
[I] I2SConfigESP32.h : 95 - number of channels: 2
[I] I2SConfigESP32.h : 96 - signal_type: Digital
[I] I2SConfigESP32.h : 98 - i2s_format: I2S_STD_FORMAT
[I] I2SConfigESP32.h : 100 - auto_clear: false
[I] I2SConfigESP32.h : 107 - buffer_count:2
[I] I2SConfigESP32.h : 108 - buffer_size:512
[I] I2SConfigESP32.h : 111 - pin_mck: 0
[I] I2SConfigESP32.h : 113 - pin_bck: 14
[I] I2SConfigESP32.h : 115 - pin_ws: 13
[I] I2SConfigESP32.h : 117 - pin_data: 15
[I] CodecWAV.h : 539 - sample_rate: 8000
[I] CodecWAV.h : 540 - channels: 2
[I] CodecWAV.h : 549 - is_streamed! because length is 2147418112
[I] StreamCopy.h : 82 - buffer_size=1073423680
Starting copier: 23899
[I] CodecWAV.h : 588 - Writing Header
[I] StreamCopy.h : 140 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops

```

**Comment by pschatzmann** [comment, marked as answer]
To my knowledge there is no such thing as a INMP443. 

If you mean the INMP441 I suggest just to read the [available documentation first](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC#inmp441---mems-microphone)!

**Reply by steenl** [reply]
Thanks, and I certainly had a typo on the INMP441.  Fixing the gain on the .wav output resolved my issue, and I will be careful from drawing false conclusions from the hex values.
FWIW this is for beehive audio analysis, taking periodic samples over time for human and spectral analysis on hive and queen health.

**Comment by pschatzmann** [comment]
Oh, that's a cool project!
